.buildkite/scripts/steps/test/scout_configs.sh

```diff
@@ -48,42 +48,43 @@ if [[ -z "$configs" && "${BUILDKITE_RETRY_COUNT:-0}" == "1" ]]; then
   fi
 fi
 
-if [ -z "$configs" ] && [ "$SCOUT_CONFIG_GROUP_KEY" != "" ]; then
+
+module_data=""
+# Download module_data if SCOUT_CONFIG_GROUP_KEY is set (needed for serverRunFlags)
+# This is required even when retrying, as we need module_data to get serverRunFlags for each config
+if [ "$SCOUT_CONFIG_GROUP_KEY" != "" ]; then
   echo "--- Downloading Scout Test Configuration"
   download_artifact scout_playwright_configs.json .
-  configs=$(jq -r '.[env.SCOUT_CONFIG_GROUP_KEY].configs[]' scout_playwright_configs.json)
+
+  # Extract module and its configs
+  module_data=$(jq -c ".[] | select(.name == env.SCOUT_CONFIG_GROUP_KEY)" scout_playwright_configs.json)
+
+  if [[ -z "$module_data" ]]; then
+    echo "Module '${SCOUT_CONFIG_GROUP_KEY}' not found in scout_playwright_configs.json"
+    exit 1
+  fi
+
+  # Extract config paths only if configs is not already set (e.g., from retry logic)
+  if [ -z "$configs" ]; then
+    # Extract config paths: process configs with their serverRunFlags directly
+    configs=$(echo "$module_data" | jq -r '.configs[].path')
+  fi
 fi
 
 if [ -z "$configs" ]; then
   echo "Unable to determine configs to run"
   exit 1
 fi
 
-# Define run modes based on group
-declare -A RUN_MODES
-RUN_MODES["platform"]="--stateful --serverless=es --serverless=oblt --serverless=security"
-RUN_MODES["observability"]="--stateful --serverless=oblt --serverless=oblt-logs-essentials"
-RUN_MODES["search"]="--stateful --serverless=es"
-RUN_MODES["security"]="--stateful --serverless=security --serverless=security-essentials --serverless=security-ease"
-
-# Define serverless-only run modes based on group
-declare -A RUN_MODES_SERVERLESS_ONLY
-RUN_MODES_SERVERLESS_ONLY["platform"]="--serverless=es --serverless=oblt --serverless=security"
-RUN_MODES_SERVERLESS_ONLY["observability"]="--serverless=oblt --serverless=oblt-logs-essentials"
-RUN_MODES_SERVERLESS_ONLY["search"]="--serverless=es"
-RUN_MODES_SERVERLESS_ONLY["security"]="--serverless=security --serverless=security-essentials --serverless=security-ease"
-
-# Determine valid run modes for the group
-if [[ -n "${SERVERLESS_TESTS_ONLY:-}" ]]; then
-  echo "--- Using serverless-only test modes (SERVERLESS_TESTS_ONLY is set)"
-  RUN_MODE_LIST=${RUN_MODES_SERVERLESS_ONLY[$group]}
-else
-  RUN_MODE_LIST=${RUN_MODES[$group]}
-fi
+# Define all possible serverless run modes
+# These are the serverless modes that can be run (excluding --stateful)
+RUN_MODES_SERVERLESS_ONLY="--serverless=es --serverless=security --serverless=oblt --serverless=oblt-logs-essentials --serverless=security-essentials --serverless=security-ease"
```

Comment:
I see we define `RUN_MODES_SERVERLESS_ONLY` but don't use it anywhere. Should we remove it?